Checker.cpp

```diff
@@ -2,21 +2,36 @@
 #include <iostream>
 using namespace std;
 
+bool isWithinRange(float value, float min, float max, std::string Message) {
+    if (value < min || value > max) {
+        cout << Message << endl;
+        return false;
+    }
+    return true;
+}
+
+bool isTemperatureOk(float temperature) {
+    return isWithinRange(temperature, 0, 45, "Temperature out of range!");
+}
+
+bool isSocOk(float soc) {
+    return isWithinRange(soc, 20, 80, "State of Charge out of range!");
+}
+
+bool isChargeRateOk(float chargeRate) {
+    return isWithinRange(chargeRate, 0, 0.8, "Charge Rate out of range!");
```

Comment:
It appears that , you assumed chargeRate min > 0 to reuse isWithinRange !

Reply:
resolved